packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts

```diff
@@ -504,7 +518,13 @@ export class Messenger {
                     responseCode,
                     codeReferenceCount: codeReference.length,
                     totalNumberOfCodeBlocksInResponse: await this.countTotalNumberOfCodeBlocks(message),
-                })
+                }
+
+                this.telemetryHelper.recordAddMessage(triggerPayload, promptAnswer)
+
+                if (message.length) {
+                    this.telemetryHelper.recordMessageReceived(triggerPayload, promptAnswer)
```

Comment:
why is this needed, why not enhance the existing `recordAddMessage` which is used just above ... ?